lib/errorlogger.h

```diff
@@ -193,12 +193,21 @@ class CPPCHECKLIB ErrorLogger {
                 : fileIndex(0), line(0), col(0) {
             }
 
-            FileLocation(const std::string &file, unsigned int aline)
-                : fileIndex(0), line(aline), col(0), mOrigFileName(file), mFileName(file) {
+            FileLocation(const std::string &file, unsigned int aline):
+                fileIndex(0),
+                line(static_cast<int>(aline)),
```

Comment:
In my humble opinion, this change does not solve any problem. If there could be any bug before this change then there can still be a bug after this change. In best case you just added a pointless cast, in worst case you hide real warnings.

I do not think we should normally use `unsigned` types. I like what the google style guide says: "try to avoid unsigned types (except for representing bitfields or modular arithmetic). Do not use an unsigned type merely to assert that a variable is non-negative."

=> A line number should not be stored in an `unsigned` integer.

I would like to see that unsigned variables are converted to signed.. feel free to open pull requests for that.. but please convert only a few variables at a time we do need to be careful.

Reply:
It actually works quite well. The signature for [tr() or translate()](https://doc.qt.io/qt-5/qcoreapplication.html#translate) is char* returning a QString. The Severity class offers a to string method returning std::string.

Reply:
The warning was prolific(every use of header produced half-dozen warnings). Looking at how 'FileLocation' was being used was consistent (providing unsigned int value). Changing the variable type in the class seemed bad given the comment in the member declaration about '-1' being allowed. Simplest solution was to cast the value. I agree this setup is bad, but this is the code as it exists. Having an implicit sign conversion in a class initialization is less than ideal. I couldn't resolve this by changing the underlying type. This may be a larger matter for refactoring.

Reply:
> Changing the variable type in the class seemed bad given the comment in the member declaration about '-1' being allowed.

oh... the `line` member is signed already so that is good. Then it's just the parameter `aline` that should be changed? It sounds risky to do it here in this PR so feel free to do that in a separate pull request.

Reply:
...to better isolate the code catastrophe or otherwise revert the commit at the first sign of foolishness...
I can do that.